sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorage.java

```diff
@@ -154,30 +164,173 @@ public void recordDouble(double value, Attributes attributes, Context context) {
   }
 
   @Override
-  public MetricData collectAndReset(long startEpochNanos, long epochNanos) {
+  public MetricDescriptor getMetricDescriptor() {
+    return metricDescriptor;
+  }
+
+  /**
+   * This method leverages the `savedDeltas` and `reportHistory` to construct a final metric point
+   * to report for this synchronous instrument.
+   */
+  @GuardedBy("collectLock")
+  @Nullable
+  private MetricData buildMetricFor(
+      CollectionHandle collector, long startEpochNanos, long epochNanos) {
+    Map<Attributes, T> result = new HashMap<>();
+    // Next merge the delta w/ the last set of points.
```

Comment:
I think some comments got out of order, since "next" is now before "first". :)

Reply:
Actually..... I had "one big method" i refactored based on my ordering comments.  Will fix :)